### ISSUE TITLE
mid_registrar

### ISSUE THREAD

**Issue by viperlong** [OpenSIPS/opensips#1321]
I update my opensips to 2.3.3  and  found " contact_match_param" do not take into effect.  even the default  "rid" do not contain in the contact uri. when i restart my opensips, the main registrar can not found the registered user.

**Comment by liviuchircu** [comment]
The mid-registrar contact matching has been reworked to advertise the internally generated Contact IDs. For now, these IDs will replace the outgoing Contact URI "username" part, but there are plans to also append this information via a `;ctid=` URI parameter, see [this PR](https://github.com/OpenSIPS/opensips/issues/1318). This is next on my TODO list.

**Comment by viperlong** [comment]
Thank you Liviu。I have found my problem，it maybe  not a bug.
In opensips 2.3.2 I used mid-registrar ，and a freeswitch as a main registrar。And I don't use db mode in mid-registrar 。this is my config：
loadmodule "mid_registrar.so"
modparam("mid_registrar", "mode", 1) 
modparam("mid_registrar", "outgoing_expires", 3600)
modparam("mid_registrar", "contact_match_param", "regid")


user A can register  with mid-register  in freeswitch，and freeswitch can originate user A and &echo.  
when I restart opensips ,freeswitch can also originate user A through “regid”。everything is  fine.

when I update to 2.3.3，restart opensips make freeswitch can't found user A. 

thanks again Liviu。
In my opinion，mid-registrar is a perfect module, and solve my big problem。
